app.py

```
@@ -46,7 +46,31 @@ def ping():
 def comports():
-    return serial_connection.available_comports()
+    try:
+        with serial_lock():
+            comports = serial_connection.available_comports()
+
+        if comports:
+            print(str(comports))
+            return jsonify(comports), 200
+        else:
+            return Response("No ports present.", 204)
+    except Exception as e:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)